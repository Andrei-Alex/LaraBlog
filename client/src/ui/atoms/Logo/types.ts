export interface ILogo {
  src: string;
  width?: string;
  height?: string;
  alt?: string;
}
