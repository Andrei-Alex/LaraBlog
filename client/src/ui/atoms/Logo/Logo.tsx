import Image from 'next/image';

import { LogoContainer, ILogo } from '.';

/**
 *
 * The Logo component is a React functional component utilized for displaying a logo within a user interface.
 * It leverages the `next/image` component for optimized image handling and the styled-component `LogoContainer` for styling.
 * This component accepts several optional props to customize the logo's appearance and accessibility features.
 *
 * ## Usage
 * ```jsx
 *   <Logo src="/custom-logo.png" alt="Custom Logo" width="100px" height="50px" />
 * ```
 *
 * @component Logo
 * @param {Object} props - The props object containing the following properties:
 * @param {string} [props.src='/logo.png'] - The source URL of the logo image. Defaults to '/logo.png'.
 * @param {string} [props.alt='Logo'] - The alt text for the logo image, enhancing accessibility. Defaults to 'Logo'.
 * @param {string} [props.width] - The width of the logo. This is passed to the `LogoContainer` for styling.
 * @param {string} [props.height] - The height of the logo. This is also passed to the `LogoContainer` for styling.
 * @param {string} [props.testID='logo'] - An identifier for testing purposes. Defaults to 'logo'.
 *
 * @returns {JSX.Element} - Returns a JSX element of the `LogoContainer` wrapping the `Image` component from `next/image`.
 *
 */

function Logo({
  src = '/logo.png',
  alt = 'Logo',
  width,
  height,
  testID = 'logo',
}: ILogo) {
  return (
    <LogoContainer width={width} height={height} data-testid={testID}>
      <Image src={src} alt={alt} layout="fill" />
    </LogoContainer>
  );
}

export default Logo;
