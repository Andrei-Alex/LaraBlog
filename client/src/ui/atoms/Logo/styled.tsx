import styled from 'styled-components';

export const LogoContainer = styled.div<{ height?: string; width?: string }>`
  height: ${({ height }) => height || '60px'};
  width: ${({ width }) => width || '120px'};
  cursor: pointer;
  position: relative;
  display: inline-block;
`;
