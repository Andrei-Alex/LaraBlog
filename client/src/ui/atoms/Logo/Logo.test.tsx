import React from 'react';
import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import Logo from './Logo';

describe('Logo', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with default props', () => {
    const { getByAltText } = render(<Logo />);
    const image = getByAltText('Logo');

    expect(image).toBeInTheDocument();
  });

  it('renders with correct width and height', () => {
    const testWidth = '100px';
    const testHeight = '50px';

    render(<Logo width={testWidth} height={testHeight} testID="test-logo" />);
    const logoContainer = screen.getByTestId('test-logo');

    expect(logoContainer).toHaveStyle(`width: ${testWidth}`);
    expect(logoContainer).toHaveStyle(`height: ${testHeight}`);
  });
});
