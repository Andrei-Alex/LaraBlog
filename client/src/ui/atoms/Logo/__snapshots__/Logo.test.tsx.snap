// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container "
    data-testid="logo"
    style="width: 5rem; height: 5rem;"
  >
    <img
      alt="Logo"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      src="/next.svg"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</DocumentFragment>
`;
